### DISCUSSION TITLE
Applying traction/force boundary conditions in Peridynamics module

### DISCUSSION THREAD

**Discussion by Narayanan1999** [idaholab/moose#27762] [Q&A General]
### Check these boxes if you have followed the posting rules.

- [X] Q&A General is the most appropriate section for my question
- [X] I have consulted the posting Guidelines on the Discussions front page
- [X] I have searched the Discussions forum and my question has not been asked before
- [X] I have searched the MOOSE website and the documentation does not answer my question
- [X] I have formatted my post following the posting guidelines (screenshots as a last resort, triple back quotes around pasted text)

### Question

I was able to apply displacement boundary conditions in Peridynamics module. But I am unable to apply traction boundary conditions. As suggested in one of the previous discussion, I have tried using 'Constantrate' for force boundary condition (for simple tension test). But, as the mesh size changes, the displacement order changes significantly in such a way that I couldn't corelate the mesh size with constant force applied. I also tried applying pressure condition in NOSPD, but it shows some segmentation fault error. In Peridynamics module also, there is no example for traction bcs. 

Please help me in this. I need to know how  'Constantrate' kernel can be used for force loading. Secondly, how else a traction boundary condition can be applied (other than using Pressure BC). It would be very helpful if an input file for simple tension test can be shared.

**Comment by GiudGiud** [comment]
@hchen139 @bwspenc

**Comment by hchen139** [comment]
I think you can applied traction boundary conditions in the PD module by either converting the traction to total force and distribute the force to nodes on the boundary and use the Constant Rate option, or construct fictitious finite elements for the boundary and use NeumannBC to apply as if you were doing FEA. For Constant Rate, you will need to update the value for different mesh density. For constructing fictitious finite element for PD, there are examples in the /tests/mesh directory.
